### ISSUE TITLE
Mixed tiles and wrong locations

### ISSUE THREAD

**Issue by tebben** [pauldendulk/Mapsui#126]
Self defined TileLayers getting mixed up when adding and drawn on the wrong locations, happens with WPF and SKIA rendering. Problem does not not seem to occur when adding 2 OSM layers. Also tested with 2 different layers instead of 2 identical urls. http://geodata.nationaalgeoregister.nl/tiles/service/tms/1.0.0/brtachtergrondkaart@EPSG:28992@jpg and http://geodata.nationaalgeoregister.nl/tiles/service/tms/1.0.0/brtachtergrondkaart@EPSG:28992@png

Test in Mapsui.Samples.WPF:

var map = new Map();
map.Layers.Add(new TileLayer(new GeozetTileSource()));
map.Layers.Add(new TileLayer(new GeozetTileSource()));
MapControl.Map = map;

       public class GeozetTileSource : ITileSource
        {
            public ITileProvider Provider { get; }
            public ITileSchema Schema { get; }
            public string Name { get; }

            public GeozetTileSource()
            {
                Schema = GetGeozetSchema();
                Provider = new HttpTileProvider(new TmsRequest(new Uri("http://geodata.nationaalgeoregister.nl/tiles/service/tms/1.0.0/brtachtergrondkaart@EPSG:28992@jpg"), Schema.Format));
                Name = "Test Geozet";
            }

            private TileSchema GetGeozetSchema()
            {
                var schema = new TileSchema();
                var resolutions = new[]
                {
                    3440.64,
                    1720.32,
                    860.16,
                    430.08,
                    215.04,
                    107.52,
                    53.76,
                    26.88,
                    13.44,
                    6.72,
                    3.36,
                    1.68,
                    0.84,
                    0.42
                };

                var count = 0;
                foreach (var resolution in resolutions)
                {
                    var levelId = count.ToString(CultureInfo.InvariantCulture);
                    schema.Resolutions[levelId] = new Resolution(levelId, resolution);
                    count++;
                }

                schema.Extent = new Extent(-285401.92, 22598.08, 595401.92, 903401.92);
                schema.OriginX = -285401.92;
                schema.OriginY = 22598.08;
                schema.Name = "GeozetSchema";
                schema.Format = "png";
                schema.YAxis = YAxis.TMS;
                schema.Srs = "EPSG:28992";

                return schema;
            }

            public Extent Extent => Schema.Extent;
            public Attribution Attribution => new Attribution();

            public byte[] GetTile(TileInfo tileInfo)
            {
                return Provider.GetTile(tileInfo);
            }
        }

**Comment by pauldendulk** [comment]
Thanks! I'll take a look.

**Comment by pauldendulk** [comment]
No repo so far. I now start the WPF sample with the sample map below. 

I encountered something like this using fiddler. It could also be related to caching somewhere in the network. Or even on the nationaalgeoregister server.

```
using System;
using System.Globalization;
using BruTile;
using BruTile.Tms;
using BruTile.Web;
using Mapsui.Layers;
using Mapsui.Utilities;
using Attribution = BruTile.Attribution;

namespace Mapsui.Samples.Common.Maps
{
    class GeozetSample
    {
        public static Map CreateMap()
        {
            var map = new Map();
            map.Layers.Add(CreateTileLayer());
            map.Layers.Add(CreateTileLayer());
            return map;
        }

        public static TileLayer CreateTileLayer()
        {
            return new TileLayer(new GeozetTileSource());
        }
    }

    public class GeozetTileSource : ITileSource
    {
        public ITileProvider Provider { get; }
        public ITileSchema Schema { get; }
        public string Name { get; }

        public GeozetTileSource()
        {
            Schema = GetGeozetSchema();
            Provider =
                new HttpTileProvider(
                    new TmsRequest(
                        new Uri(
                            "http://geodata.nationaalgeoregister.nl/tiles/service/tms/1.0.0/brtachtergrondkaart@EPSG:28992@jpg"),
                        Schema.Format));
            Name = "Test Geozet";
        }

        private TileSchema GetGeozetSchema()
        {
            var schema = new TileSchema();
            var resolutions = new[]
            {
                3440.64,
                1720.32,
                860.16,
                430.08,
                215.04,
                107.52,
                53.76,
                26.88,
                13.44,
                6.72,
                3.36,
                1.68,
                0.84,
                0.42
            };

            var count = 0;
            foreach (var resolution in resolutions)
            {
                var levelId = count.ToString(CultureInfo.InvariantCulture);
                schema.Resolutions[levelId] = new Resolution(levelId, resolution);
                count++;
            }

            schema.Extent = new Extent(-285401.92, 22598.08, 595401.92, 903401.92);
            schema.OriginX = -285401.92;
            schema.OriginY = 22598.08;
            schema.Name = "GeozetSchema";
            schema.Format = "png";
            schema.YAxis = YAxis.TMS;
            schema.Srs = "EPSG:28992";

            return schema;
        }

        public Extent Extent => Schema.Extent;
        public Attribution Attribution => new Attribution();

        public byte[] GetTile(TileInfo tileInfo)
        {
            return Provider.GetTile(tileInfo);
        }

    }
}
```

**Comment by pauldendulk** [comment]
Can still not reproduce. Switched my wifi on and off during panning. Added random exceptions in fetcher. Added random delays too. Noting out of the ordinary happened. I will try again during working hours. Maybe it helps if the site itself is under stress.

**Comment by pauldendulk** [comment]
Closing because we can not reproduce it. We assume it is not a Mapsui issue. If someone sees this issue again please reopen and describe your case.

**Comment by pauldendulk** [comment]
Could reproduce this on another network. The cause is still a mystery. Some of the facts:

- It occurs with the GeozetSample posted above. It has two of the same layers. In my test it was the first sample. It occurs immediately on the first zoom in.
- So far the problem did not show with two different layers.
- The problem does not show when testing at my home (although that was on an another (older) machine).
- The problem goes away when Fiddler is open.

**Comment by pauldendulk** [comment]
The problem disappears when I add a custom tile fetcher method to the HttpTileProvider like this:

``` c#
        HttpClient _httpClient = new HttpClient();
        private byte[] FetchTile(Uri arg)
        {
            
            return  _httpClient.GetByteArrayAsync(arg).Result;
        }

```
The reappears  again if I make the HttpClient static:
``` c#
        static HttpClient _httpClient = new HttpClient();
        private byte[] FetchTile(Uri arg)
        {
            
            return  _httpClient.GetByteArrayAsync(arg).Result;
        }
```

**Comment by pauldendulk** [comment]
The current fetcher in BruTile is the older WebClient. It is not defined as static. I see nothing so far that would explain this behavior.

**Comment by pauldendulk** [comment]
The PCL profile of BruTile allows use of HttpClient. So this could be a possible solution.

**Comment by pauldendulk** [comment]
Fixed by upping to BruTile 0.20.0

**Comment by pauldendulk** [comment]
In another context the problem re-occurred again, even with HttpClient. More research was done by @tebben. 

- A minimal application without Mapsui/BruTile or any other nuget dependencies showed the problem too. 
- A virtual machine with a clean install of Windows did not have the problem.
- When webroot (security software) was installed the problem re-occured again!

This seems a very serious bug that is probably not restricted to .NET.
